Install Grav (CMS) + plugins

capsulecorplab opened the pull request:
closes #15 

## Testing procedure
1. (In a Linux terminal) clone or download `fleet-workspace-image` repo (if not already cloned)
```sh
cd ~
git clone https://github.com/startrekthefleet/fleet-workspace-image.git
```
2. change directory to `fleet-workspace-image` repo
```sh
cd ~/fleet-workspace-image
```
3. In a text editor (s.a., VS Code), change `:main` to `:pr-28` or `:install-grav` on image name in `docker-compose.yml`
  a. Run `code .` to open VS Code
  b. Open `docker-compose.yml` to edit the image name
  c. Save changes to `docker-compose.yml`
4. Pull docker image with `docker-compose pull` (this may take a few minutes)
5. Spin up docker image with `docker-compose up`
6. Log into kasm session by going to `https://localhost:6901` in a web browser and enter credentials
7. Create and serve new grav project
  a. open a terminal window from within kasm session
  b. create a Grav project directory with `mkdir MyGravProject`
  c. change directory into Grav project with `cd MyGravProject`
  d. run Grav server with `grav serve`

capsulecorplab committed:
install grav (CMS) tools

installs grav v1.7.42.3 from source tarball from install_files

capsulecorplab committed:
update node tools

- update node_version to 18.5.0
- update npm_version to 8.12.1
- update npx_version to 8.12.1
- update yarn_version to 3.5.0
- update generator_code_version to 1.7.5
- update yo_version to 4.3.1

capsulecorplab committed:
globally install Grav tools

- extract grav tarball into /opt
- create symlink for grav tools in /usr/local/bin
- test grav version

capsulecorplab committed:
install gravity

capsulecorplab committed:
update gravity version v0.1.0-beta

capsulecorplab committed:
Install DevTools plugin for Grav

capsulecorplab committed:
update gravity version to v0.1.0-rc1

capsulecorplab committed:
update Grav version to 1.7.43

capsulecorplab committed:
Install Admin plugin for Grav

capsulecorplab committed:
add web dev tools to s/w list in README

capsulecorplab committed:
add gravity to web dev tools in s/w list in README

capsulecorplab commented:
Admin panel works as of https://github.com/startrekthefleet/fleet-workspace-image/pull/28/commits/186560312312a496d0ea899c08e0a379f806919c
![Screenshot from 2024-01-11 11-59-15](https://github.com/startrekthefleet/fleet-workspace-image/assets/14095576/0bf38146-9a5e-4f81-8876-e086ac7d7d2c)
![Screenshot from 2024-01-11 11-59-27](https://github.com/startrekthefleet/fleet-workspace-image/assets/14095576/88326a69-355b-46e0-a768-2d1065565894)

sarasage commented:
Possibly relevant to the CSS issue:

https://stackoverflow.com/questions/63000219/setting-up-scss-compilation-in-a-custom-theme-in-grav-cms

capsulecorplab committed:
npm install -g sass@1.70.0

sarasage commented:
Install grav-plugin-sitemap:
https://github.com/getgrav/grav-plugin-sitemap

How to access sitemap on site running on local host:
http://127.0.0.1:8000/sitemap

capsulecorplab committed:
update node package list in s/w list in README

capsulecorplab committed:
update Grav version to 1.7.44

capsulecorplab committed:
Install Sitemap plugin for Grav

capsulecorplab committed:
Install Telephasic theme for Grav

capsulecorplab committed:
add Sitemap plugin and Telephasic theme for Grav in s/w list in README

capsulecorplab committed:
updates to CI

- printout disk space
- add workaround to increase disk space
- update docker/build-push-action to v5
- ignore publishing to GHCR on PR
